### PR TITLE
z_bg_mizu_movebg and object_mizu_objects

### PR DESCRIPTION
**Pull request by mzxrules**
i only memorized the vanilla water temple layout

### CODE REVIEW

File: spec [hunk 1]
```diff
@@ -1029,8 +1029,6 @@ endseg
 beginseg
     name "ovl_Bg_Mizu_Movebg"
     include "build/src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.o"
-    include "build/data/overlays/actors/z_bg_mizu_movebg.data.o"
-    include "build/data/overlays/actors/z_bg_mizu_movebg.bss.o"
     include "build/data/overlays/actors/z_bg_mizu_movebg.reloc.o"
```
Comment (AngheloAlf) [inline spec:1032]: Remember to use the generated reloc instead.
```suggestion
#ifdef NON_MATCHING
    include "build/src/overlays/actors/ovl_Bg_Mizu_Movebg/ovl_Bg_Mizu_Movebg_reloc.o"
#else
    include "build/data/overlays/actors/z_bg_mizu_movebg.reloc.o"
#endif
```

File: assets/xml/objects/object_mizu_objects.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<Root>
+    <File Name="object_mizu_objects" Segment="6">
+        <DList Name="gObjectMizuObjectsMovebgDL_000190" Offset="0x0190"/>
+        <Collision Name="gObjectMizuObjectsMovebgCol_0003F0" Offset="0x03F0"/>
```
Comment (AngheloAlf) [inline assets/xml/objects/object_mizu_objects.xml:4]: Could you try to add a name to dlists and collisions?
You can ask in the discord server if you need help.

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 3]
```diff
@@ -21,21 +38,353 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
...
+f32 D_8089EB40[] = {
+    -115.200005,
+    -115.200005,
+    -115.200005,
+    0.0f,
+};
+Gfx* D_8089EB50[] = { gObjectMizuObjectsMovebgDL_000190, gObjectMizuObjectsMovebgDL_000680,
+                      gObjectMizuObjectsMovebgDL_000C20, gObjectMizuObjectsMovebgDL_002E10,
+                      gObjectMizuObjectsMovebgDL_002E10, gObjectMizuObjectsMovebgDL_002E10,
+                      gObjectMizuObjectsMovebgDL_002E10, gObjectMizuObjectsMovebgDL_0011F0 };
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:51]: Could you add a trailing comma and run `format.sh` again?

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 4]
```diff
@@ -21,21 +38,353 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
...
+
+CollisionHeader* D_8089EB70[] = { &gObjectMizuObjectsMovebgCol_0003F0, &gObjectMizuObjectsMovebgCol_000998,
+                                  &gObjectMizuObjectsMovebgCol_000ED0, &gObjectMizuObjectsMovebgCol_003590,
+                                  &gObjectMizuObjectsMovebgCol_003590, &gObjectMizuObjectsMovebgCol_003590,
+                                  &gObjectMizuObjectsMovebgCol_003590, &gObjectMizuObjectsMovebgCol_0015F8 };
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:56]: Same here for the trailing comma.

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 5]
```diff
@@ -21,21 +38,353 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
...
+               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0 * 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
+                                        this->scrollAlpha1));
+
+    gSPSegment(POLY_OPA_DISP++, 0x09,
+               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0 * 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
+                                        this->scrollAlpha2));
+
+    gSPSegment(POLY_OPA_DISP++, 0x0A,
+               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0 * 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
+                                        this->scrollAlpha3));
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:377]: Do you really need the `0 * 0`?
```suggestion
    gSPSegment(POLY_OPA_DISP++, 0x08,
               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
                                        this->scrollAlpha1));

    gSPSegment(POLY_OPA_DISP++, 0x09,
               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
                                        this->scrollAlpha2));

    gSPSegment(POLY_OPA_DISP++, 0x0A,
               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, frames * 1, 0, 32, 32, 1, 0, 0, 32, 32, 0, 0, 0,
                                        this->scrollAlpha3));
```

**Review by AngheloAlf — CHANGES_REQUESTED** [review]
Good job!
Only a few minors issues.

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c (hunk 5 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:377]: TwoTexScrolls being what they are, probably

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 6]
```diff
@@ -21,21 +38,353 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
     (ActorFunc)BgMizuMovebg_Update,
     (ActorFunc)BgMizuMovebg_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Movebg/func_8089DC30.s")
 
+f32 D_8089EB40[] = {
```
Comment (louist103) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:42]: `static` for all of these.

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 7]
```diff
@@ -21,21 +38,353 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
...
+                                         (WATER_TEMPLE_WATER_F1_Y - WATER_TEMPLE_WATER_B1_Y) * (255 - 160));
+    } else {
+        this->scrollAlpha3 = 160;
+    }
+    this->scrollAlpha4 = this->scrollAlpha3;
+}
+
+#ifdef NON_MATCHING
+void func_8089E318(BgMizuMovebg* this, GlobalContext* globalCtx) {
+    WaterBox* waterBoxes; // sp3C
```
Comment (louist103) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:239]:
```suggestion
    WaterBox* waterBoxes;
```

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.h [hunk 8]
```diff
@@ -6,9 +6,19 @@
...
-    /* 0x014C */ char unk_14C[0x3C];
+    /* 0x0000 */ DynaPolyActor dyna;
+    /* 0x0164 */ BgMizuMovebgActionFunc actionFunc;
+    /* 0x0168 */ f32 unk_168;
+    /* 0x016C */ s32 scrollAlpha1;
+    /* 0x0170 */ s32 scrollAlpha2;
+    /* 0x0174 */ s32 scrollAlpha3;
+    /* 0x0178 */ s32 scrollAlpha4;
+    /* 0x017C */ u8 unk_17C;
+    /* 0x0180 */ Gfx* unk_180;
```
Comment (louist103) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.h:20]: I'd probably just name this DList or something

**Review by louist103 — CHANGES_REQUESTED** [review]
Looks good. Just some small things.

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 9]
```diff
@@ -1,15 +1,32 @@
...
 
 #define FLAGS 0x00000010
 
 #define THIS ((BgMizuMovebg*)thisx)
 
+#define BgMizuMovebgGetType(params) (((u16)(params) >> 0xC) & 0xF)
+#define BgMizuMovebgGetFlags(params) ((u16)(params)&0x3F)
+#define BgMizuMovebgGetPathId(params) (((u16)(params) >> 0x8) & 0xF)
+#define BgMizuMovebgGetPointId(params) ((u16)(params)&0xF)
+#define BgMizuMovebgGetSpeed(params) (((u16)(params) >> 0x4) & 0xF)
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:19]: we arent doing macros for these until we agree on a standardize way to do them. it is planned for a documentation pass, which will be really soon as we are almost done with actors

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 10]
```diff
@@ -21,21 +38,357 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
...
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Movebg/func_8089DC30.s")
 
+static f32 D_8089EB40[] = {
+    -115.200005,
+    -115.200005,
+    -115.200005,
+    0.0f,
+};
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:47]: id prob remove the comma on this, it looks short enough. unless you think its something modders would want to add more numbers to

**Review by fig02 — APPROVED** [review]
changes made

File: src/overlays/actors/ovl_Bg_Mizu_Bwall/z_bg_mizu_bwall.c [hunk 11]
```diff
@@ -378,31 +380,28 @@ void BgMizuBwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
...
 
-    if (waterLevel < -15.0f) {
+    if (waterLevel < WATER_TEMPLE_WATER_F1_Y) {
         this->scrollAlpha1 = 255;
-    } else if (waterLevel < 445.0f) {
-        alphaMod = ((waterLevel - -15.0f) / (445.0f - -15.0f)) * 95.0f;
-        this->scrollAlpha1 = 255 - alphaMod;
+    } else if (waterLevel < WATER_TEMPLE_WATER_F2_Y) {
+        this->scrollAlpha1 = 255 - (s32)((waterLevel - WATER_TEMPLE_WATER_F1_Y) / (WATER_TEMPLE_WATER_F2_Y - WATER_TEMPLE_WATER_F1_Y) * (255 - 160));
+
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mizu_Bwall/z_bg_mizu_bwall.c:390]:
```suggestion
```

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 12]
```diff
@@ -21,21 +38,351 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
...
+                D_8089EE40 &= ~1;
+            }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mizu_Movebg/BgMizuMovebg_Destroy.s")
+s32 func_8089E108(Path* pathList, Vec3f* pos, s32 pathId, s32 pointId) {
+    Path* path = pathList;
+    Vec3s* point;
+    path += pathId;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:196]:
```suggestion

    path += pathId;
```

File: src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c [hunk 13]
```diff
@@ -21,21 +38,351 @@ const ActorInit Bg_Mizu_Movebg_InitVars = {
...
+
+void BgMizuMovebg_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgMizuMovebg* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void BgMizuMovebg_Draw(Actor* thisx, GlobalContext* globalCtx2) {
+    BgMizuMovebg* this = THIS;
+    GlobalContext* globalCtx = (GlobalContext*)globalCtx2;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mizu_Movebg/z_bg_mizu_movebg.c:357]:
```suggestion
    GlobalContext* globalCtx = globalCtx2;
```

**Review by Roman971 — COMMENTED** [review]
Just a few things. Also I think the formatter needs to be run on most of the changed files

**Comment by fig02** [comment]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**